django/thunderstore/repository/api/v1/views/listing_index.py

```diff
@@ -0,0 +1,39 @@
+import json
+
+from django.http import HttpResponse
+from django.shortcuts import get_object_or_404, redirect
+from drf_yasg.utils import swagger_auto_schema  # type: ignore
+from rest_framework.request import Request
+from rest_framework.views import APIView
+
+from thunderstore.community.models import Community
+from thunderstore.repository.models import APIV1ChunkedPackageCache
+
+
+class PackageListingIndex(APIView):
+    """
+    Return a blob file containing URLs to package listing chunks.
+    Client needs to gunzip and JSON parse the blob contents.
+
+    /c/{community_id}/api/v1/package-listing-index/
+    """
+
+    @swagger_auto_schema(
+        tags=["api"],
+        auto_schema=None,  # Hide from API docs for now.
+    )
+    def get(self, request: Request, community_identifier: str):
+        community = get_object_or_404(
+            Community.objects.listed(),
+            identifier=community_identifier,
+        )
+        cache = APIV1ChunkedPackageCache.get_latest_for_community(community)
+
+        if cache:
+            return redirect(request.build_absolute_uri(cache.index.data_url))
+
+        return HttpResponse(
+            json.dumps({"error": "No cache available"}),
+            status=503,
+            content_type="application/json",
+        )
```

Comment:
Not a blocker, just a note

REST Framework should have a shorthand for this which is just `Response(data, status=503)` without the need to json encode or set the content type manually, but I'm fine with this implementation too.

Reply:
Less imports and less code, so changed as suggested.